Public/Src/Engine/Cache/Fingerprints/BondExtensions.cs

```diff
@@ -131,7 +131,7 @@ public static FileMaterializationInfo ToFileMaterializationInfo(this BondFileMat
             Contract.Requires(bondInfo != null);
 
             return new FileMaterializationInfo(
-                new FileContentInfo(bondInfo.Hash.ToContentHash(), bondInfo.Length),
+                new FileContentInfo(bondInfo.Hash.ToContentHash(), FileContentInfo.LengthAndExistence.CreateWithCombinedValue(bondInfo.Length)),

```

Comment:
>Length [](start = 135, length = 6)

So this isn't really Length anymore as it has other data encoded in it. Should you rename the field? Same applies to the data passed into other callsites of CreateWithCombinedValue()